### PR TITLE
Update data model nav

### PR DESCRIPTION
**Pull request by kdoh**
Changes to the data model to make it easier to view all your segments and metrics across databases or tables.

Screen Shot 2020-06-09 at 11 50 22 AM

Things to un-snarl

- Now that there are specific listing pages for each, I think we should move the segment and metric descriptions into the listing endpoint instead of having to have metadata down on the client to generate them.
- You can now create segments and metrics from the listings but as a result we don't have a way to pre-populate the segment / metric editor with the table you hit `Add segment` from. We'll need to add a data picker to both.

ToDos

- [x] Add metric list view
- [x] Clean up style of database selector on the db view
- [x] Make tab selected state correspond to the url
- [x] List view empty states

### CODE REVIEW

**Comment by robdaemon** [comment]
@kdoh ping me on Monday and I can help out with the backend portion of this!

**Comment by paulrosenzweig** [comment]
I added a data selector on top to populate the metric/segment forms and filter the list. There are still some things I need to fix though (e.g. retiring segments/metrics).

**Comment by mazameli** [comment]
It wasn't clear to me that I needed to select a table on the list view in order to make the "New segment" button clickable. I think this incorrectly conflates filtering the list view with picking the data to base a new segment on. 

image

My strong assumption was that I'd be able to pick the table from this first element in the query builder here where it says `Orders`:

image

Going back to the list view, I also strongly assumed that I'd be able to view _all_ of my instance's segments or metrics from these list views without _having_ to filter by a table first; that's one of the benefits of this reorg, as otherwise you have to hunt and peck through your schema to figure out which tables have any segments or metrics. To be explicit, I think that table filtering element should stay; I just don't think that using it should be required.

image

**Comment by robdaemon** [comment]
@mazameli - I had a similar thought to you on that - Maybe clicking "new segment" shows the table chooser?

I guess having a way to filter by table (or sort by table) would be good for customers that have a lot of metrics or segments too?

**Comment by paulrosenzweig** [comment]
@mazameli I made those updates.

> My strong assumption was that I'd be able to pick the table from this first element in the query builder

For metric/segment creation, there's a data selector in that spot in the QB. For existing metrics/segments, it's locked and you can't click it. That should prevent breaking existing queries.

---

> I also strongly assumed that I'd be able to view all of my instance's segments or metrics from these list views without having to filter by a table first

I like this change, but now we probably need a way to deselect a chosen table and return to an unfiltered list. I'm guessing an "x" to the left of the table name is the way to go?

**Comment by mazameli** [comment]
@paulrosenzweig Sweet. The new-segment and new-metric data selection feels exactly right to me now.

> I like this change, but now we probably need a way to deselect a chosen table and return to an unfiltered list. I'm guessing an "x" to the left of the table name is the way to go?

Yeah, ideally I'd like to follow our dashboard filters UI pattern:

image

image

**Comment by paulrosenzweig** [comment]
@mazameli I couldn't get the legend on `<FieldSet>` to work without ugly jumping around as it was added/removed, so I took it out:

gif

File: frontend/src/metabase/query_builder/components/AggregationPopover.jsx [hunk 1]
```diff
@@ -250,14 +250,15 @@ export default class AggregationPopover extends Component {
...
-        : [];
+    const metrics = tableMetadata.metrics
+      ? tableMetadata.metrics.filter(metric =>
+          showMetrics
+            ? !metric.archived ||
+              (selectedAggregation && selectedAggregation.id === metric.id)
+            : // GA metrics are more like columns, so they should be displayed even when showMetrics is false
+              metric.googleAnalyics,
+        )
+      : [];
```
Comment (paulrosenzweig) [inline frontend/src/metabase/query_builder/components/AggregationPopover.jsx:261]: @tlrobinson in lieu of a larger review, what do you think about putting this here? Previously we were creating a `new Metadata` that filtered out non-Google Analytics metrics. That felt ugly and I figured the PartialQueryBuilder should use the full Metadata and filter it when displaying. Is there a better spot for this logic?

Definitely feel free to comment on whatever else here if you want, but there's a big diff and most of this PR is moving things around and small refactors.

Comment (tlrobinson) [reply frontend/src/metabase/query_builder/components/AggregationPopover.jsx:261]: Yeah I think this is better than a custom Metadata

**Review by paulrosenzweig — APPROVED** [review]
This is somewhat a self-approval, but three of us have worked on this together. When the tests pass, I'll merge so we can get the 36 RC out.

**Pull request merged**